ppb/features/collision.py

```diff
@@ -0,0 +1,108 @@
+"""
+A collision subsystem with various colliders available.
+"""
+from itertools import product
+
+from ppb.sprites import BaseSprite
+from ppb.systemslib import System
+
+
+class CollidedWithMixin:
+    """
+    Provides a mechanism for CollisionCheckerSystem to inform sprites when
+    they collide with something.
+
+    This is a mixin added to CanCollideCircleMixin and
+    CanCollideSquareMixin. You do not need to to add it to your sprites if
+    you are using one of them.
+    """
+
+    def collided_with(self, other):
```

Comment:
This should probably look like an event handler: `on_collided_with(self, event, signal)`.

The subsystem is going to skip the event plumbing because it's not strictly necessary, but it'll make it more clear what's going on.